Pipeline Execution Service

TimothyTickle opened the pull request:
Summary of major points of discussion: 
- The original charter was split into pipelines development and pipelines execution (here).
- Conversations over how to best communicate what pipelines are of concern with this service, obviously not all data pipelines in the DCP as they would be ubiquitous.
- Formatting was updated.

~~**December 8**: Last call for community reviews~~
~~**February 26**: Last call for oversight reviews~~

TimothyTickle committed:
Added template.

TimothyTickle committed:
First Draft Charter

TimothyTickle committed:
Used full name for GG

mweiden commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,80 @@
+
+# Project Name
+HCA Data Processing Service
```
You can simply make the heading `# HCA Data Processing Service`

TimothyTickle replied:
no problem, adding to update.

sampierson replied:
This is a completely meaningless name.  We couldn't stick with `Secondary Analysis Service`?  The fact you have to put that in parens after the name points to its lack of utility.

TimothyTickle replied:
Secondary Analysis Service was also deemed confusing and meaningless so I am going with what, I believe, the content/UX team are going with. Very much open to another suggestion.

sampierson replied:
I came up with a better idea below "Pipeline Execution Service".  Resolving this conversation

mweiden commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,80 @@
+
+## Description
+
+The data processing service is a centralized space to perform all transformative functionality to scientific data ingested into the Human Cell Atlas Data Coordination Platform (HCA DCP).
```
nit: `[perform functionality]`? How about something like `execute transformation functions on ...`?

TimothyTickle replied:
The updated language on the use of the word transformation; will update shortly.

mweiden commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,80 @@
+
+## Definitions
+
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
```
* `project, but`
* nit: Many of the items in lists here are not full sentences. I'd remove the period from those that aren't. Apply this rule consistently throughout the doc.

TimothyTickle replied:
thanks!

mweiden commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,80 @@
+
+## In-scope
+
+* Scoping or review of scoping for new pipelines.  
```
* Like above ⬆️, remove periods from non-sentences in a list.
* Preferably, write all bullets in active voice: `Create, maintain, and operate ...`. Apply this consistently throughout the doc.

TimothyTickle replied:
thank you for the feedback.

mweiden commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,80 @@
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
+__HCA Analysis Working Group (AWG):__ A group of domain specialists that advise on and monitor scientific choices made in the Human Cell Atlas Project.  
+__HCA DCP Governance Group:__ A working group of the Human Cell Atlas project that advises the DCP, assuring alignment of DCP development efforts with the scientific project it supports.  
+__On-boarding:__ The process of taking an existing pipeline and engineering it to work in the HCA DCP.  
```
nit: [Onboarding](https://en.wikipedia.org/wiki/Onboarding) is not hyphenated.

TimothyTickle replied:
thanks!

mweiden commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,80 @@
+* Outreach to developers on pipeline development in the HCA DCP.  
+* Suggestion but not selection of new standardized pipelines for the HCA DCP.  
+
+### Scientific "guardrails" [Optional]
```
Remove `[Optional]`

TimothyTickle replied:
Good catch.

bkmartinjr commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,80 @@
+
+* Scoping or review of scoping for new pipelines.  
+* Creation of all standardized pipelines for scientific data.  
+* Review, on-boarding, and potential creation of community pipelines.  
```
To enable third parties to create community pipelines, there needs to be a modest amount of support infra (minimally documentation; maximally tooling or support).   Suggest that at least the minimum enabling "stuff" be included.   

Proposed addition:   "Tools which support the creation of community pipelines - may include software, documentation or support."

("may" indicates that it is optional - just leaving open the possibility that the team decides any or all of these are needed)

kbergin replied:
Great point, @bkmartinjr  - I think Tim is planning to make a separate charter for Unity, the service we are creating to enable more community creation of pipelines. I do think that documentation / support around the pipelines we run can fall under Data Processing Service charter, though. Thoughts?

bkmartinjr replied:
I'm in favor of having it in this charter, as it makes it clear that the community needs to be enabled by the team (within reason - obviously it is not a replacement for Unity)

TimothyTickle replied:
Thanks, this language is placed in the HCA Data Processing Pipelines charter, new version coming shortly.

jonahcool replied:
Tend to agree with Bruce here and think that some sort of pointer would be helpful. Given that this charter is a overarching statement and relates to more specific charters (eg pipelines, unity) indicating that support for enabling community contribution is a key feature of the top level goals.

kbergin commented:
@mweiden thanks for all the comments - just wanted to note I've seen them and will get to updating soon!

lauraclarke commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,80 @@
+
+## In-scope
+
+* Scoping or review of scoping for new pipelines.  
```
I wonder if like some of the other charters (Ingest service and data portal most notably) it is worth spliting this list into the data driven objectives like the review of new pipelines, onboarding of pipelilnes and discover of benchmark data from the operational objectives like optimization and operation

kbergin replied:
👍 on it

lauraclarke commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,80 @@
+* Establishing and operating automated practices around scientific and engineering testing of pipelines.  
+* Outreach to users on the usability of data in downstream applications.  
+* Outreach to developers on pipeline development in the HCA DCP.  
+* Suggestion but not selection of new standardized pipelines for the HCA DCP.  
```
There is an absence of connection to the metadata standards.

The data processing team needs to work with the community to provide feedback if the metadata standards for raw data provide the required information to enable processing and work with the metadata team and data consumers to ensure the metadata provided about pipelines and the analysis results meet the needs of downstream users

TimothyTickle replied:
Language added to the upcoming/new HCA Data Processing Pipelines charter.

lauraclarke commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,80 @@
+* Creation, maintenance, and operation of the HCA DCP pipeline execution environment.  
+* Investigating and identifying data for pipeline benchmarking.  
+* Initiation and management of the execution of all data processing pipelines (community or standardized) that run in the HCA DCP.  
+* Evaluation and adoption of file formats (and associated standards) for outputs of pipelines.  
```
Is it valuable to call out to FAIR standards here with respect to file formats, ensuring the data is accessible with standard tools and interoperable with downstream processes

TimothyTickle replied:
I am not sure where this goes now that the expression matrix service exists. Arguably zarr is not FAIR compared to other file formats but the DCP as a whole is FAIR given we can produce many file formats for users (given prioritization by the expression matrix team). So BAMs are FAIR but zarr I think is not (currently) but performant and translated downstream to something that is FAIR. The DCP as a whole should produce FAIR data but not sure if pipelines can be pulled out from the whole system in this case. Thoughts?

lauraclarke commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,80 @@
+* Educational training or user assistance on how to perform data analysis with outputs of pipelines.  
+* Recommendations or demonstrations of downstream tool use beyond initial data loading.  
+* Novel methodological development is not performed by the data processing team unless it is essential to the pipeline, it does not existent in community tools, and it represents minimal effort to implement.  
+* Software engineering 3rd party tools (although may happen in high value community tools, is not a requirement of the service and left to the discretion of the team).  
```
I would leave out the caveat here otherwise people will view this statement as a negotiating position rather than something your team can decide for yourselves.

TimothyTickle replied:
Took your advice ;-)

kbergin committed:
Update charter.md

Responding to all comments and splitting this charter into two - one for the service and one for pipelines, which will be in a separate PR.

kbergin commented:
Hello all - I updated the charter to reflect all of your comments (thanks!) and split the charter into two, one for data processing service, and I will open a branch shortly for the pipelines. 

cc @lauraclarke @bkmartinjr @mweiden

kbergin committed:
Rename charter.md to DataProcessingService.md

Updating name

brianraymor commented on charters/DataProcessingService/DataProcessingService.md:
```diff
@@ -0,0 +1,45 @@
+# [Data Processing Service](mailto:pipelines-team@data.humancellatlas.org)
```
Per the process, this file must be named charter.md.

TimothyTickle replied:
thanks!

brianraymor committed:
Editorial changes to Communication section

brianraymor committed:
Editorial updates for slack channel "style"

brianraymor committed:
Editorial updates to Github repositories section

Now with lists ...

brianraymor commented on charters/DataProcessingService/DataProcessingService.md:
```diff
@@ -0,0 +1,51 @@
+# [Data Processing Service](mailto:pipelines-team@data.humancellatlas.org)
+
+## Description
```
Change all instances of _data processing service_ to _Data Processing Service_

The Data Processing service ~is a service~ 

What does _executes automated transformation_ mean? 
How does _scientific data_ relate to the Human Cell Atlas?
What are _scientific pipelines_?
What is an _instantiation_ of the DCP?

TimothyTickle replied:
Changed data processing service -> Data Processing Service.
Changed the language around the use of the word transformation, please check in the new version.
I am not clear on "How does scientific data relate to the Human Cell Atlas?" If it is still a concern in the newer version (when posted shortly) please let me know.
Moving away from "scientific pipelines" and towards "data processing pipelines".
Changed "instantiation" with "instance", I hope that clarifies.

brianraymor commented on charters/DataProcessingService/DataProcessingService.md:
```diff
@@ -0,0 +1,51 @@
+## In-scope
+* Creating and maintaining the Data Processing Service
+* Operating the instantiation of the HCA DCP Data Processing Service  
+* Initiation and management of the execution of all data processing pipelines that run in the HCA DCP  
```
Is there a difference between _scientific_ pipelines and _data processing_ pipelines?

TimothyTickle replied:
No, language is moving over to data processing pipelines.

brianraymor commented on charters/DataProcessingService/DataProcessingService.md:
```diff
@@ -0,0 +1,51 @@
+* Initiation and management of the execution of all data processing pipelines that run in the HCA DCP  
+* Documentation on the Data Processing Service and guides on using the service 
+* Establishing and operating automated practices around engineering testing of pipelines in the HCA DCP 
+* Creating and outputting metadata on the transformations done to the data in accordance with the Metadata Schema
```
**DCP** Metadata Schema to be more specific?

TimothyTickle replied:
I used HCA DCP, let me know what you think in the next version.

mweiden commented:
Looks good to me other than the comments @brianraymor left. 👍

brianraymor committed:
Editorial updates to Slack Channels

Adopt best practices for slack channel format

ambrosejcarr commented on charters/DataProcessingService/DataProcessingService.md:
```diff
@@ -0,0 +1,51 @@
+# [Data Processing Service](mailto:pipelines-team@data.humancellatlas.org)
+
+## Description
+The data processing service is a service that executes automated transformation to scientific data through cloud-based execution of scientific pipelines. It can be used in any instantiation of the Data Coordination Platform (DCP). 
```
"Transformative" is emphasized here. Maybe its meaning should be defined. What does this imply about this service? 

Should this be:
```diff
+The data processing service is a service that executes automated transformation of scientific data through cloud-based execution of scientific pipelines
-The data processing service is a service that executes automated transformation to scientific data through cloud-based execution of scientific pipelines
```

lauraclarke replied:
+1 for to to of

TimothyTickle replied:
added a definition and reworded.

ambrosejcarr commented on charters/DataProcessingService/DataProcessingService.md:
```diff
@@ -0,0 +1,51 @@
+The data processing service is a service that executes automated transformation to scientific data through cloud-based execution of scientific pipelines. It can be used in any instantiation of the Data Coordination Platform (DCP). 
+
+## Objectives
+The data processing service aims to develop the infrastructure to run transformative scientific analysis pipelines reliably and at-scale for an instantiation of the DCP. The team working on the service will also operate the infrastructure for the Human Cell Atlas. 
```
I got a bit confused by "for an instantiation of the DCP". It makes me think you're developing this for a specific DCP. I would scratch that part of the objective: 

The data processing service aims to develop the infrastructure to run transformative scientific analysis pipelines reliably and at-scale for ~an instantiation of~ the DCP. The team working on the service will also operate the infrastructure for the Human Cell Atlas.


I think this conveys the same idea. Please let me know if I've misinterpreted this.

TimothyTickle replied:
Updated.

lauraclarke commented on charters/DataProcessingService/DataProcessingService.md:
```diff
@@ -0,0 +1,51 @@
+* Creating and outputting metadata on the transformations done to the data in accordance with the Metadata Schema
+
+## Out-of-scope
+* Standing up or operating other instantiations of the Data Processing Service
```
I assume this is because you will operate the version of the data processing service for the HCA instance of the DCP but not anyone else's data processing service using the code created by this service.

I think the language throughout the charter could be clearer. This is the first time such an explicit statement is made whereas elsewhere it feels more implied.

TimothyTickle replied:
We are thinking about reuse and what are our obligations to reuse. Documentation and script for deployment are things we would work towards making and improving but we would not deploy or operate. I think it is important we think about reuse and point to the DCP as an open source and reusable project.

TimothyTickle committed:
Updating charter given community review.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,65 @@
+[Rex Wang](mailto:chengche@broadinstitute.org)  
+
+## Communication
+
```
Since we're no longer shipping colored boxes, (there's no blue box charter), I would recommend renaming _mint*_ to avoid use of the DCP secret decoder ring -or- change the description for the slack channels not to use _mint*_.

To match the style of other charters, could you copy and paste this into the **Communications** section, replacing the current text?

```
### Mailing List
Team email: pipelines-team@data.humancellatlas.org

### Slack Channels
 * [HumanCellAtlas/mintteam](https://humancellatlas.slack.com/messages/mintteam): team discussions, a place to directly communicate to the team, who works on the service and the pipelines, as needed.
 * [HumanCellAtlas/mintteam-ci](https://humancellatlas.slack.com/messages/mintteam-ci): mint team continuous integration status
```

kbergin replied:
Mint team isn't named mint team because of colored boxes, it's just our team name at the Broad. We could make separate channels for user facing purposes i.e. #dataprocessingservice , but the #mintteam channel is meant for our inter-team discussion.

TimothyTickle replied:
This is interesting because I think a lot of the teams named their channels for their product and but we did using our internal team name (the color naming is a false positive as Kylee mentioned; we actually have completely unrelated internal green and red teams as well). I can definitely see it would be difficult to know which channel to go to in the slack (this charter of course does help and guide). As Kylee mentioned, happy to make a #dataprocessingservice channel and list it here as an outward facing channel.

kbergin replied:
Yes - I'd recommend renaming the secondary analysis one though, not creating a new one.

TimothyTickle replied:
Adding updated names in the charter, will update actual slack around Friday.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,65 @@
+
+## Description
+
+The Data Processing Service is a service that executes data processing pipelines on DCP data in the Data Storage Service. It can be used in any instance of the Data Coordination Platform (DCP). 
```
What is _any instance_ of the DCP?

Do you want to include a Definition for pipelines?

TimothyTickle replied:
Remove the "any instance" language.

Added a definition.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,65 @@
+
+## Github repositories
+
+[Cromwell Tools](https://github.com/broadinstitute/cromwell-tools): Tools (Python API) for working with the Cromwell workflow engine  
```
Could _Cromwell_ be removed or clarified in the repo description? For example:

`_Tools (Python API) for working with [Cromwell](https://github.com/broadinstitute/cromwell) - a scientific workflow engine designed for simplicity and scalability_`

TimothyTickle replied:
This is really helpful, happy to pull that in directly.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,65 @@
+
+## Objectives
+
+The Data Processing Service aims to develop the infrastructure to run pipelines reliably and at-scale for the DCP. The team working on the service will also operate the infrastructure for the Human Cell Atlas. 
```
_operate the infrastructure_ -> _operate the *pipeline* infrastructure_ 

Editorial:
_The Data Processing Service ~aims to~ develops the infrastructure to run pipelines reliably and at-scale for the DCP. The team ~working on~ implementing the service ~will~ also operate**s** the pipeline infrastructure for the Human Cell Atlas._

TimothyTickle replied:
updated.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,65 @@
+* Standing up or operating other instances of the Data Processing Service for other projects  
+
+## Milestones and Deliverables
+
```
Is it possible to offer more details (Q1, H2). Otherwise, these read as aspirational for 2019 and 2020.

TimothyTickle replied:
I am struggling to be more specific in this timeframe. I feel even these could change as our current GA and project requirements are rapidly evolving. For example, I have had conversations today about bundle structure which should potentially be added here. I am not sure how to be more specific in 2020 other than general themes; we could be more specific and only speak about 2019 Q1&2 but then the charter would need to be renewed/updated at that point. Happy to talk offline, I may be missing something.

mweiden reviewed (changes requested):
Pardon, rereading this after the split from the pipelines charter, I have to agree with @sampierson: this charter is too general as written. Reading it as an outside observer, it could be confused with other data consuming software projects.

Please add details that differentiate the data processing service from other data consumer applications.

Some possible improvements along these lines could be
* calling out that this service supports the chartered `Data Processing Pipelines` project explicitly
* offering a more detailed description of pipelines as suggested by @brianraymor

TimothyTickle committed:
Added definition and editorial changes.

brianraymor commented:
@mweiden, @ambrosejcarr - would you review the update and approve/reject?

mweiden commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,69 @@
+
+## Definitions
+
+__pipeline:__ A collection of one or more functional tasks that operate on input data and, from that, make an output that is transformed input data, or features generated to interpret the input data.
```
@TimothyTickle thank you for adding this. Still, this definition could apply to other software projects in DCP. Take the data browser, for example:

* `A collection of one or more functional tasks that operate on input data` - the data browser backend includes tasks for assembling bundles into specimens and indexing them
* `make an output that is transformed input data` - transforms bundles into specimen records for the data browser
* `or [make] features generated to interpret the data` - one could argue that facets used for searching the data are extracted "features"

This will likely be confusing to uninitiated readers.

This definition/charter should provide sufficient information to answer the question: what makes data processing pipelines in the green box unique from other software projects in DCP?

ambrosejcarr replied:
Per Matt's comment, I think this definition could be clarified by answering why these pipelines are important or what they typically contain.

TimothyTickle replied:
I am not sure I am intending this definition to be used as it is. The definition is simply defining a "pipeline", it is not meaning to answer the question "what makes data processing pipelines in the green box unique" but simply what is a "pipeline". It is simply a definition of a particular word in the charter.

Given the current charter, the description of the service would be a better place to describe the data processing service and associated pipelines as a whole and I think it does differentiate the functionality it is concerned with from others by specifying it works on the scientific data and performs analysis associated with science. This should differentiate itself from other services in the system as none should be doing that. None should be generating features used for scientific interpretation including the browser but using data ingested into the system or data generated from the pipelines.

I will add an aspect of "batch" to the definition of the pipeline to try to characterize more toward traditional data processing pipelines but I worry that is overfitting given solutions that may come, but if ends up being true, we can update the definition. I will also update the description a little to make sure it is clear how the service differentiates itself.

ambrosejcarr commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,69 @@
+
+## In-scope
+
+* Creating and maintaining the Data Processing Service  
```
This section seems redundant with the Objectives to me.

TimothyTickle replied:
removed.

ambrosejcarr commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,69 @@
+
+* Operating the instance of the HCA DCP Data Processing Service
+* Initiation and management of the execution of all data processing pipelines that run on scientific data in the HCA DCP  
+* Establishing and operating automated practices around engineering testing of pipelines executed in the HCA DCP Date Processing Service   
```
```suggestion
* Establishing and operating automated practices around engineering testing of pipelines executed in the HCA DCP Data Processing Service
```

TimothyTickle replied:
thanks, updated.

ambrosejcarr commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,69 @@
+
+## Out-of-scope
+
+* Standing up or operating other instances of the Data Processing Service for other projects  
```
Is this needed? The charter is in the `humancellatlas` github org, so I can't envision anyone thinking that through this charter, you're agreeing to process data or operate a service for another group.

TimothyTickle replied:
I believe there will be a time where the products and projects are less confounded and this will be important but I am happy to remove this, for now. The charters can be updated of course.

TimothyTickle committed:
Updated charter definition and minor edits.

TimothyTickle committed:
Minor edit

TimothyTickle committed:
removed an underscore

TimothyTickle committed:
Minor edits.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+## Communication
+
+[Team email](mailto:pipelines-team@data.humancellatlas.org): pipelines-team@data.humancellatlas.org 
```
This creates redundant mailto: links. Could you change to: `Team email: pipelines-team@data.humancellatlas.org`

TimothyTickle replied:
Done.

brianraymor commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,63 @@
+ 
+__HumanCellAtlas#data-pipelines-team:__ team discussions, a place to directly communicate to the team that works on the service and pipelines  
+__HumanCellAtlas#pipelines-ci:__ Data Processing Service continuous integration status  
+
```
Could you update the slack channels to follow the best practice:
`[HumanCellAtlas/data-pipelines-team](https://humancellatlas.slack.com/messages/data-pipelines-team)` and `[HumanCellAtlas/pipelines-ci](https://humancellatlas.slack.com/messages/pipelines-ci)`

TimothyTickle replied:
Updated to current slack channels

sampierson commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+## Objectives
+
+The Data Processing Service develops the infrastructure to run pipelines reliably and at-scale for the DCP. The team implementing the service also operates the pipeline infrastructure for the Human Cell Atlas project. 
```
"The Data Processing Service develops the infrastructure" doesn't scan.  The team develops the service and infrastructure.  The service doesn't develop infrastructure.

TimothyTickle replied:
Updated, will push in a moment.

sampierson commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,63 @@
+# [Data Processing Service](mailto:pipelines-team@data.humancellatlas.org)
+
```
This is a terrible name for this charter.  "Data Processing" is so nebulous as to be meaningless.  All DCP components "process data".   It's barely better calling it "The Computing Service".  It doesn't say anything about analysis or pipelines, the two key things that this service does.  I see that development of pipelines has been given its own charter. So this is really the "Pipeline Execution Service".

TimothyTickle replied:
Updated, will push in a moment.

sampierson commented on charters/DataProcessingService/charter.md:
```diff
@@ -0,0 +1,63 @@
+__pipeline:__ A collection of one or more functional tasks that operate on input data and, from that, make an output that is transformed input data, or derive features often used to interpret the input data. In a high throughput setting, these tasks are often automated to be performed in a batch setting.
+
+## In-scope
+
```
This In-scope section is pretty light.  Sure, yes, Ops and Docs, but what about development, maintenance and improvement of the service itself?  What about the pipelines - which pipelines will the service run?  Only Standardized Pipelines?  Will we make a "best-effort" attempt to run anything that a third party comes up with?

TimothyTickle replied:
We had something about maintenance and improvement but the feedback we got was that such a thing was obviously in scope and need not be said. I am happy to add it back in if you feel it needs to be there. This charter will have to be updated once we better understand community pipelines.

TimothyTickle committed:
Updated charter name and clarified Objective

TimothyTickle committed:
Updating email

rexwangcc commented on charters/PipelineExecutionService/charter.md:
```diff
@@ -1,12 +1,12 @@
-# [Data Processing Service](mailto:pipelines-team@data.humancellatlas.org)
+# [Pipeline Execution Service](mailto:pipelines-team@data.humancellatlas.org)
```
Could you please also update the title of this PR to reflect this change?

TimothyTickle replied:
Done

rexwangcc replied:
Thanks!

rexwangcc commented on charters/PipelineExecutionService/charter.md:
```diff
@@ -51,8 +51,8 @@ __2020:__ Generalized and reusable service infrastructure
-__HumanCellAtlas#data-pipelines-team:__ team discussions, a place to directly communicate to the team that works on the service and pipelines  
-__HumanCellAtlas#pipelines-ci:__ Data Processing Service continuous integration status  
+[HumanCellAtlas/data-pipelines-team](https://humancellatlas.slack.com/messages/data-pipelines-team): team discussions, a place to directly communicate to the team that works on the service and pipelines  
+[HumanCellAtlas/data-pipelines-ci](https://humancellatlas.slack.com/messages/data-pipelines-ci): Data Processing Service continuous integration status  
```
`Data Processing Service` -> `Pipeline Execution Service` or explicitly indicating that this channel will cover both data pipelines tests and execution infra tests

TimothyTickle replied:
I added language to hopefully make it clear the channel is for both if you want them separate that is fine as well, just let me know.

rexwangcc replied:
Keeping it together makes more sense to me, thanks!

rexwangcc commented on charters/PipelineExecutionService/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+### Operations
+
+* Operating the instance of the HCA DCP Data Processing Service
```
not sure: `Data Processing Service` -> `Pipeline Execution Service` (consequences of changing a title 😄 )

TimothyTickle replied:
nice catch

rexwangcc commented on charters/PipelineExecutionService/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+* Operating the instance of the HCA DCP Data Processing Service
+* Initiation and management of the execution of all data processing pipelines that run on scientific data in the HCA DCP  
+* Establishing and operating automated practices around engineering testing of pipelines executed in the HCA DCP Data Processing Service   
```
same, `Pipeline Execution Service`

TimothyTickle replied:
nice catch, sorry

rexwangcc commented on charters/PipelineExecutionService/charter.md:
```diff
@@ -0,0 +1,63 @@
+### Documentation
+
+* Creating and outputting metadata on the operations applied to the data in accordance with the HCA DCP Metadata Schema  
+* Documentation on the Data Processing Service and guides on using the service  
```
Same, `Data Processing Service` -> `Pipeline Execution Service`

TimothyTickle replied:
got it

rexwangcc commented on charters/PipelineExecutionService/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+* Creating and outputting metadata on the operations applied to the data in accordance with the HCA DCP Metadata Schema  
+* Documentation on the Data Processing Service and guides on using the service  
+* Creating documentation and scripting around the deployment of the Data Processing Service
```
`Data Processing Service` -> `Pipeline Execution Service`

TimothyTickle replied:
thanks!

rexwangcc commented on charters/PipelineExecutionService/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+## Milestones and Deliverables
+
+__2018:__ Create operational and secure infrastructure, add ability to throttle and queue  
```
optional: ability to throttle and queue the executions of data processing pipelines/workflows

TimothyTickle replied:
added

rexwangcc commented on charters/PipelineExecutionService/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+## Github repositories
+
+[Cromwell Tools](https://github.com/broadinstitute/cromwell-tools): Tools (Python API) for working with the [Cromwell workflow engine](https://github.com/broadinstitute/cromwell) - a scientific workflow engine designed for simplicity and scalability  
```
```suggestion
[Cromwell Tools](https://github.com/broadinstitute/cromwell-tools): A collection of Python clients and accessory scripts for interacting with the [Cromwell workflow execution engine](https://github.com/broadinstitute/cromwell) - a scientific workflow engine designed for simplicity and scalability  
```

TimothyTickle replied:
Updated to your language.

rexwangcc replied:
Thanks

rexwangcc reviewed (commented):
Left some comments, thanks for the work!!

TimothyTickle committed:
Updated service name in doc

rexwangcc reviewed (approved):
Thanks!

samanehsan commented on charters/PipelineExecutionService/charter.md:
```diff
@@ -0,0 +1,63 @@
+
+## Definitions
+
+__pipeline:__ A collection of one or more functional tasks that operate on input data and, from that, make an output that is transformed input data, or derive features often used to interpret the input data. In a high throughput setting, these tasks are often automated to be performed in a batch setting.
```
One suggestion on wording: I think changing "make an output that is transformed input data" to "transform the input data" makes this sentence easier to understand

TimothyTickle replied:
added

samanehsan commented on charters/PipelineExecutionService/charter.md:
```diff
@@ -0,0 +1,63 @@
+[Falcon](https://github.com/HumanCellAtlas/falcon): Queueing system that (after launching) throttles and inititates workflows  
+[Lira](https://github.com/HumanCellAtlas/lira): Listens to storage service notifications and launches workflows  
+[Pipeline Tools](https://github.com/HumanCellAtlas/pipeline-tools): Contains Data Coordination Platform adapter pipelines and associated tools  
+[Secondary Analysis](https://github.com/HumanCellAtlas/secondary-analysis): Integration test scripts and other automation for the Pipeline Execution Service (contains all tickets for the prioritized work for the implementation team)  
```
In addition to the repositories listed here, we recently added the [Secondary Analysis Deploy](https://github.com/HumanCellAtlas/secondary-analysis-deploy) repository that contains the deployment configuration and scripts for the Pipeline Execution Service

TimothyTickle replied:
added

TimothyTickle committed:
Added new repo and updated better language on pipeline definition.

rexwangcc committed:
Update charter.md
